designer/client/conditions/select-condition-helpers.ts

```diff
@@ -0,0 +1,29 @@
+import { ConditionData } from "./SelectConditions";
+
+// return whether the current condition has an object value
+export const isObjectCondition = (condition: ConditionData) => {
+  return typeof condition.value !== "string";
+};
+
+// return whether the current condition value has the conditionName property
+export const hasConditionName = (condition: any) => {
+  return !!condition?.conditionName;
+};
+
+// return whether the current condition has a nested condition as a value
+export const hasNestedCondition = (condition: ConditionData) => {
+  return condition.value.conditions?.find?.(hasConditionName);
```

Comment:
let's just make sure if the optional chain returns undefined at any point, OR nothing can be found, return false.


`return condition.value.conditions?.find?.(hasConditionName) ?? false` 

`??` is the nullish coalesce operator. if a value is null or undefined, return the value on the righthand side. in this case false.